api/src/main/java/com/example/api/service/user/UserService.java

```diff
@@ -147,4 +157,52 @@ public String getProfessorRegisterToken() throws WrongUserTypeException {
         log.info("Professor {} fetch ProfessorRegisterToken", user.getEmail());
         return professorRegisterToken.getToken();
     }
+
+    public void deleteProfessorAccount(String email) throws WrongUserTypeException {
+        User user = getCurrentUser();
+        User professor = userRepo.findUserByEmail(email);
+        userValidator.validateProfessorAccount(professor, email);
+        userValidator.validateProfessorAccount(user, user.getEmail());
+
+        changeUserForActivitiesAndAdditionalPoints(user, professor);
+        userRepo.delete(user);
+    }
+
+    public void deleteStudentAccount() throws WrongUserTypeException {
```

Comment:
Wsm pytanie: czy gdy usuniemy konto studenta, który wysłał aktywność do sprawdzenia prowadzącemu to czy ta aktywność zostanie wyświetlona u prowadzącego ? Jeżeli tak to czy powinna być bo sprawdzanie aktywności studentowi, który usunął konto jest chyba bez sensu

Reply:
wszystkie resulty wraz z usunieciem konta studenta tez sie usuwaja